68 project gallery

ryanyae opened the pull request:
## Description

Created a new component that is a carousel for projects

## Updates

created a new component and added only instance of it in the Correlation's project

## Checklist

- [x] Ran `npm run lint:fix` to format code
- [x] Requested at least one reviewer
- [x] Connected PR to Trello ticket (steps below)
- [ ] Addressed code review comments
- [ ] Gained at least one approval for your PR

## Connecting your PR to the corresponding ticket:
- Click on the "GitHub" button found on the right of your Trello ticket
- Choose "Attach PR" from the dropdown list
- Link your GitHub account (if you haven't done so already)
- Navigate to`VCL-content-platform` and choose your PR from the dropdown.

ryanyae commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Project Gallery](https://trello.com/c/C0xaQdw5/68-project-gallery)

ryanyae commented:
This is one of the questions that I had, I had tried to make a 3D carousel and noticed that there was a lot of hardcode and ended up with this implementation. If this is not the type of carousel that is desired I could try and go implement a 3D version of it.

ElisaHu1 commented:
![image](https://github.com/UBC-VCL/VCL-content-platform/assets/104804980/eece7264-7d10-4059-82d2-29102d8e64b3)
instead of changing the package.lock.json, you should change pakage.json instead for this PR

ElisaHu1 reviewed (commented):
looks amazing, are you able to change the scale of the image and leave some paddings like the pic in figma
<img width="322" alt="image" src="https://github.com/UBC-VCL/VCL-content-platform/assets/104804980/8fe7a882-c9eb-422f-be1c-2e1832b7d0bd">
so the images does not look distorted

ryanyae commented:
@ElisaHu1, right I for sure am able to. Earliest I can get you these changes maybe tomorrow afternoon.

svolterra commented on package.json:
```diff
@@ -37,7 +38,7 @@
   },
   "scripts": {
-    "start": "export SET NODE_OPTIONS=--openssl-legacy-provider && react-app-rewired start",
+    "start": "set NODE_OPTIONS=--openssl-legacy-provider && react-app-rewired start",
```
Hmm interesting that package.json and package-lock.json are both in the .gitignore file yet they were still committed 🤔

ryanyae replied:
Yea, I am kind of struggling with this fix. Been searching on how to fix this problem and none seem work. I will try to figure out a fix to this problem, hopefully soon.

svolterra commented on src/components/ProjectGallery/ProjectGallery.tsx:
```diff
@@ -0,0 +1,133 @@
+        }
+    }
+
+    // This is the autoscrolling feature
```
Amazing addition! Works really well too :) Great work!

svolterra commented on src/components/ProjectGallery/ProjectGallery.tsx:
```diff
@@ -0,0 +1,133 @@
+                <div id='container-titles'>
+                    {
+                        /* 
+                            The titles below should not be hardcoded like this, but should be instead inside a prop
```
100% agree with this comment :)

svolterra reviewed (commented):
Really really great work Ryan! The gallery looks incredible and I absolutely love the auto-scrolling feature 😁 I do agree with Elisa on the photo padding, but otherwise looks amazing :) Phenomenal work :)

svolterra reviewed (commented):
Thank you for addressing the code review comments! I was just wondering if it would be possible to move the image title and the image itself slightly to the right such that their positions align more with the **Gallery** title? They appear to be a little off-centered at the moment 🤔 

<img width="1167" alt="Screen Shot 2023-07-03 at 8 43 30 PM" src="https://github.com/UBC-VCL/VCL-content-platform/assets/71746168/8332d6c7-0060-4a42-8263-21cea78a9b39">

Otherwise looks amazing 💯 🏆 👏

ryanyae commented:
> Thank you for addressing the code review comments! I was just wondering if it would be possible to move the image title and the image itself slightly to the right such that their positions align more with the **Gallery** title? They appear to be a little off-centered at the moment 🤔
> 
> <img alt="Screen Shot 2023-07-03 at 8 43 30 PM" width="1167" src="https://user-images.githubusercontent.com/71746168/250736792-8332d6c7-0060-4a42-8263-21cea78a9b39.png">
> Otherwise looks amazing 💯 🏆 👏

I just fixed it with my most recent push. It was just a little CSS component that needed to be fixed

ElisaHu1 reviewed (approved):
autoscrolling stoped working with the new commit, besides that, I am good with this pr

ryanyae commented:
@ElisaHu1 @svolterra I made changes to the branch after I fixed what Elisa had mentioned and I do not see the commit here. I hope they fix by itself, but if it doesn't I could bring it up in tomorrows meeting.
